src/Fixer/ControlStructure/SingleExpressionPerLineFixer.php

```diff
@@ -0,0 +1,398 @@
+                    $fixSwitchCases,
+                    [':', ';'],
+                    $this->getSwitchColonIndexes($tokens)
+                );
+
+                $index = $until + $added;
+                $tokenAddedCount += $added;
+
+                continue;
+            }
+
+            // @TODO: remove defined condition when PHP 8.0+ is required
+            if (\defined('T_MATCH') && $tokens[$prevIndex]->isGivenKind(T_MATCH)) {
+                $until = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_PARENTHESIS_BRACE, $index);
+                $tokenAddedCount += $this->processBlock($tokens, $index, $until, $fixControlStructures);
+
+                $index = $tokens->getNextTokenOfKind($index, ['{']);
+                $until = $tokens->findBlockEnd(Tokens::BLOCK_TYPE_CURLY_BRACE, $index);
+                $added = $this->processBlock($tokens, $index, $until, $fixMatch, [',']);
+
+                $index = $until + $added;
+                $tokenAddedCount += $added;
+            }
+        }
+
+        if ($shouldAddNewLine) {
+            $tokenAddedCount += $this->addNewLineBeforeIfNecessary($tokens, $end + $tokenAddedCount, -1);
+        }
+
+        return $tokenAddedCount;
+    }
+
+    /**
+     * @return array<int>
+     */
+    private function getSwitchColonIndexes(Tokens $tokens): array
+    {
+        if (null === $this->switchColonIndexes) {
+            $colonIndexes = [];
+
+            /** @var SwitchAnalysis $analysis */
+            foreach (ControlCaseStructuresAnalyzer::findControlStructures($tokens, [T_SWITCH]) as $analysis) {
+                $default = $analysis->getDefaultAnalysis();
+
+                if (null !== $default) {
+                    $colonIndexes[] = $default->getColonIndex();
+                }
+
+                foreach ($analysis->getCases() as $caseAnalysis) {
+                    $colonIndexes[] = $caseAnalysis->getColonIndex();
+                }
+            }
+
+            $this->switchColonIndexes = $colonIndexes;
+        }
+
+        return $this->switchColonIndexes;
+    }
+
+    private function addNewLineAfterIfNecessary(Tokens $tokens, int $index, int $extraIndentation = 0): int
+    {
+        $next = $tokens->getNextMeaningfulToken($index);
+        if ($tokens->isPartialCodeMultiline($index, $next - 1)) {
+            return 0;
+        }
+
+        return $this->addNewLineAfter($tokens, $index, $extraIndentation);
+    }
+
+    private function addNewLineBeforeIfNecessary(Tokens $tokens, int $index, int $extraIndentation = 0): int
+    {
+        $previous = $tokens->getPrevMeaningfulToken($index);
+        if ($tokens->isPartialCodeMultiline($previous, $index)) {
+            return 0;
+        }
+
+        return $this->addNewLineAfter($tokens, $previous, $extraIndentation);
+    }
+
+    private function addNewLineAfter(Tokens $tokens, int $index, int $extraIndentation = 0): int
+    {
+        $indent = $this->getIndentation($tokens, $index, $extraIndentation);
```

Comment:
I would remove indentation handling and leave that to `statement_indentation` and `array_indentation` rules.

Reply:
I took the indentation handling from MethodArgumentSpaceFixer.

If the method argument is doing it, I was thinking it could be nice do it too.
But I'd like to avoid indeed.

If I want to rely on statement_indentation and array_indentation, I need them to be perfect.
And https://github.com/PHP-CS-Fixer/PHP-CS-Fixer/issues/7884 is one blocker for example.

Reply:
IMO we should rather rework `method_argument_space` to *not* handle indentation too (maybe with a deprecation path though).

For the record, you should be able to work around #7884 by fixing the wrong indent manually. The rule should not "unfix" it on subsequent run.

Reply:
> For the record, you should be able to work around #7884 by fixing the wrong indent manually. The rule should not "unfix" it on subsequent run.

Yes, but the rule won't be popular (especially for big code base) if you have to fix every single multi-line controle structure indentation after a run... It's definitely a no-go IMHO.